src/new/wallet.js

```diff
@@ -290,6 +290,25 @@ class HathorWallet extends EventEmitter {
    * @return {Promise} Promise that resolves when transaction is sent
    **/
   sendTransaction(address, value, token) {
+    const ret = this.prepareTransaction(address, value, token);
+
+    if (ret.success) {
+      return transaction.sendTransaction(ret.data, this.pinCode);
+    } else {
+      return Promise.reject(ret.message);
+    }
```

Comment:
As tx has already been prepared, can't we just `return this. sendPreparedTransaction(ret.data)`?

Reply:
You mean change this whole `if ... else` block for this? If it's that, we can't.

We could change

`return transaction.sendTransaction(ret.data, this.pinCode);`
to
`return this. sendPreparedTransaction(ret.data);`

I didn't feel it would make much difference. But I can change it, if you prefer.

Reply:
Ah, I thought we would be able to replace the whole `if/else`, didn't notice we were checking the result of `this.prepareTransaction`.

Still, I think using `return this.sendPreparedTransaction(ret.data);` is better. Otherwise we'll be calling `transaction.prepareData` twice, once in `this.prepareTransaction` and also in `transaction.sendTransaction`.

Reply:
Done!